src/control_structures.js

```diff
@@ -0,0 +1,120 @@
+/** Create an action that chooses one of two actions to run, depending on
+ * a predicate applied to the seed value.
+ *
+ * @param {function} predicate  A function that returns either true or
```

Comment:
style (applies several times in this file):
`function` isn't a primitive so it should have initial caps. Also a dash between the name and description
```suggestion
 * @param {Function} predicate - A function that returns either true or
```